### PR TITLE
librbd: fix potential image rm failure due to parent missing

### PR DESCRIPTION
**Pull request by Songweibin**
Signed-off-by: songweibin <song.weibin@zte.com.cn>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dillaman** [comment]
@Songweibin Any clue how the parent was deleted? Did you delete the pool or something?

**Comment by Songweibin** [comment]
> @Songweibin Any clue how the parent was deleted? Did you delete the pool or something?

Yes, the parent volume pool was accidentally deleted.

File: src/librbd/image/RemoveRequest.cc [hunk 1]
```diff
@@ -187,8 +187,10 @@ void RemoveRequest<I>::handle_detach_child(int r) {
   if (r < 0) {
     lderr(m_cct) << "failed to detach child from parent: "
                  << cpp_strerror(r) << dendl;
-    send_close_image(r);
-    return;
+    if (r != -ENOENT) {
```
Comment (dillaman) [inline src/librbd/image/RemoveRequest.cc:190]: I'd prefer to hide this in the actual `DetachChildRequest` state machine. For example, if this was failing in `DetachChildRequest<I>::clone_v2_child_detach` when it attempts to create the `IoCtx` for the child, or `DetachChildRequest<I>::handle_clone_v2_get_snapshot` when it attempts to read the snapshot, handle the `-ENOENT` there.

**Comment by Songweibin** [comment]
@dillaman  Updated. Thank you.

**Comment by dillaman** [comment]
jenkins test make check

**Comment by dillaman** [comment]
jenkins test docs

**Pull request merged**